### PR TITLE
✨ Upgrade scaffolds to use go 1.23

### PR DESCRIPTION
**Pull request by camilamacedo86**
Upgrade scaffolds from go version 1.22 to 1.23

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v5
         with:
-          go-version: '1.22'
+          go-version: '1.23'
```
Comment (mateusoliveira43) [inline .github/workflows/codeql.yml:28]: would be better to use `go-version-file: go.mod`?

Comment (camilamacedo86) [reply .github/workflows/codeql.yml:28]: Would you like to push a PR with this change?
Yes, that it would be better if work out in this case

Comment (mateusoliveira43) [reply .github/workflows/codeql.yml:28]: yes, can work on that next week

**Review by mateusoliveira43 — APPROVED** [review]

**Comment by camilamacedo86** [comment]
/skip pull-kubebuilder-e2e-k8s-1-32-0

not working yet

File: testdata/project-v4/go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module sigs.k8s.io/kubebuilder/testdata/project-v4
 
-go 1.22.0
+go 1.23.0
+
+godebug default=go1.23
```
Comment (ldemailly) [inline testdata/project-v4/go.mod:5]: curious what does that do when you also have go 1.23.0 ?

Comment (camilamacedo86) [reply testdata/project-v4/go.mod:5]: Why would we need to set the patch for debug?

Comment (ldemailly) [reply testdata/project-v4/go.mod:5]: I should have clarified: I am asking why you added these `godebug` lines with the default that happens to exactly match the go version in the go line; my understanding is that is a "noop" but I'd love to learn why it isn't

(I am not talking about patchlevel at all)

Comment (ldemailly) [reply testdata/project-v4/go.mod:5]: opened https://github.com/kubernetes-sigs/kubebuilder/issues/4786 - ty

**Pull request merged**